runhouse/resources/hardware/cluster.py

```diff
@@ -1082,47 +1062,22 @@ def _start_or_restart_helper(
         resync_rh: Optional[bool] = None,
         restart_ray: bool = True,
         restart_proxy: bool = False,
+        parallel: bool = True,
     ):
         if not self.is_up():
             raise ConnectionError(
                 f"Could not reach {self.name} {self.head_ip}. Is cluster up?"
             )
 
-        image_secrets, image_env_vars = self._sync_image_to_cluster()
-
-        # If resync_rh is not explicitly False, check if Runhouse is installed editable
-        local_rh_package_path = None
-        if resync_rh is not False:
-            local_rh_package_path = Path(
-                importlib.util.find_spec("runhouse").origin
-            ).parent
-
-            installed_editable_locally = (
-                not _rh_install_url
-                and local_rh_package_path.parent.name == "runhouse"
-                and (local_rh_package_path.parent / "setup.py").exists()
-            )
-
-            if installed_editable_locally:
-                logger.debug("Runhouse is installed locally in editable mode.")
-                resync_rh = True
-            else:
-                # We only want this to be set if it was installed editable locally
-                local_rh_package_path = None
+        if resync_rh:
+            self.image.install_packages(["ray", "pip:runhouse"])
+            logger.debug("Finished syncing Runhouse to cluster.")
 
-        # If resync_rh is still not confirmed to happen, check if Runhouse is installed on the cluster
-        if resync_rh is None:
-            return_codes = self.run_bash_over_ssh(["runhouse --version"], node="all")
-            if return_codes[0][0][0] != 0:
-                logger.debug("Runhouse is not installed on the cluster.")
-                resync_rh = True
+        image_secrets, image_env_vars = self._sync_image_to_cluster(parallel=parallel)
 
         if resync_rh:
-            self._sync_runhouse_to_cluster(
-                _install_url=_rh_install_url,
-                local_rh_package_path=local_rh_package_path,
-            )
-            logger.debug("Finished syncing Runhouse to cluster.")
+            # remove the runhouse install so it doesn't get reinstalled on restart
```

Comment:
should this be `if not rsync_rh`?

Reply:
No - this is specifically if we had rsync_rh enabled and therefore added some items to the image. Here we're cleaning them up by deleting them.